crates/miden-objects/src/block/block_body.rs

```diff
@@ -0,0 +1,157 @@
+use alloc::vec::Vec;
+
+use miden_core::utils::{
+    ByteReader,
+    ByteWriter,
+    Deserializable,
+    DeserializationError,
+    Serializable,
+};
+
+use crate::block::{BlockAccountUpdate, OutputNoteBatch, ProposedBlock};
+use crate::note::Nullifier;
+use crate::transaction::OrderedTransactionHeaders;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct BlockBody {
+    /// Account updates for the block.
+    updated_accounts: Vec<BlockAccountUpdate>,
+
+    /// Note batches created by the transactions in this block.
+    output_note_batches: Vec<OutputNoteBatch>,
+
+    /// Nullifiers created by the transactions in this block through the consumption of notes.
+    created_nullifiers: Vec<Nullifier>,
+
+    /// The aggregated and flattened transaction headers of all batches in the order in which they
+    /// appeared in the proposed block.
+    transactions: OrderedTransactionHeaders,
+}
+
+impl BlockBody {
+    pub fn new(
+        updated_accounts: Vec<BlockAccountUpdate>,
+        output_note_batches: Vec<OutputNoteBatch>,
+        created_nullifiers: Vec<Nullifier>,
+        transactions: OrderedTransactionHeaders,
+    ) -> Self {
```

Comment:
```suggestion
    /// Returns a new [`BlockBody`] instantiated from the provided components.
    ///
    /// # Warning
    ///
    /// This constructor does not do any validation, so passing incorrect values may lead to later
    /// panics.
    pub fn new_unchecked(
        updated_accounts: Vec<BlockAccountUpdate>,
        output_note_batches: Vec<OutputNoteBatch>,
        created_nullifiers: Vec<Nullifier>,
        transactions: OrderedTransactionHeaders,
    ) -> Self {
```

I'd rename this to `new_unchecked` because the consistency of accounts, nullifiers or output notes with respect to the transaction headers is not checked.